aten/src/ATen/core/op_registration/test_helpers.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <ATen/core/Tensor.h>
+#include <ATen/core/dispatch/Dispatcher.h>
+#include <ATen/Functions.h>
+#include <ATen/core/ivalue.h>
+
+template<class... Inputs>
+std::vector<c10::IValue> makeStack(Inputs&&... inputs) {
+  return {std::forward<Inputs>(inputs)...};
+}
+
+at::Tensor dummyTensor(c10::TensorTypeId dispatch_key) {
+  auto* allocator = at::getCPUAllocator();
+  int64_t nelements = 1;
+  auto dtype = caffe2::TypeMeta::Make<float>();
+  auto storage_impl = c10::make_intrusive<c10::StorageImpl>(
+    dtype,
+    nelements,
+    allocator->allocate(nelements * dtype.itemsize()),
+    allocator,
+    /*resizable=*/true);
+  return at::detail::make_tensor<c10::TensorImpl>(storage_impl, dispatch_key, false);
+}
+
+template<class... Args>
+std::vector<c10::IValue> callOp(const c10::OperatorHandle& op, Args... args) {
+  auto stack = makeStack(std::forward<Args>(args)...);
+  auto kernel = c10::Dispatcher::singleton().lookup(op, &stack);
```

Comment:
it's not about this diff - but it'd be really good to see an example of how this would be used in something like JIT interpreter. Right now it creates a new instance on each invocation. So if you want to dispatch once at the beginning (to reuse the kernel cache), we'd need to statically provide information about arguments in a different form (available through Node*) or dispatch each time at runtime but not re-create kernel cache if we dispatched to the same instance. Current API just creates new kernel cache instance each time

Reply:
There's a test case for this, see `givenKernelWithCache_whenCalled_thenCacheIsHandledCorrectly`.